platform/darwin/src/MGLBaseStyleLayer.h

```diff
@@ -0,0 +1,26 @@
+#import <Foundation/Foundation.h>
+
+@interface MGLBaseStyleLayer : NSObject
+
+@property (nonatomic, assign, getter=isVisible) BOOL visible;
+
+/**
+ The maximum zoom level on which the layer gets parsed and appears on.
+ */
+@property (nonatomic, assign) float maxZoom;
+
+/**
+ The minimum zoom level on which the layer gets parsed and appears on.
+ */
+@property (nonatomic, assign) float minZoom;
+
+/**
+ Updates the layers layout and paint properties.
+ 
+ @discussion The properties are recalculated upon managing sources and
+ layers but in order to update a layout or paint property you have to call
+ this method manually.
+ */
+- (void)update;
```

Comment:
As tail work, let’s consider a way to update immediately by default but offer a block-based batch updating mechanism for use cases that involve changing multiple properties in tandem.